tests/routes/test_tasks.py

```diff
@@ -1,107 +1,196 @@
-    #     "status": "pending",
-    #     "detail": "None"
+            ]
+        }
+        patch_resp = client.patch(f"/tasks/{task['task_id']}", json=patch_payload)
+        assert patch_resp.json()["status"] == "success"
+
+    version_resp = client.get(f"/meta/{dataset}/{version}")
+    assert version_resp.json()["data"]["status"] == "saved"
+
+    asset_resp = client.get(f"/meta/{dataset}/{version}/assets/{asset_id}")
+    assert asset_resp.json()["data"]["status"] == "saved"
+
+    # The following will fail until creation of auxiliary assets is working
+
+    # # Now that the default asset is saved we can create a non-default
+    # # asset, which is handled slightly differently. In particular if
+    # # creation of the auxiliary asset fails the version remains in the
+    # # "saved" state. Let's verify that.
+    # asset_payload = {
+    #     "asset_type": "Vector tile cache",
+    #     "asset_uri": "http://www.slashdot.org",
+    #     "is_managed": False,
+    #     "creation_options": {
+    #         "zipped": False,
+    #         "src_driver": "GeoJSON",
+    #         "delimiter": ",",
+    #     },
     # }
-    # patch_resp = client.patch(f"/meta/tasks/{new_task_id}", data=json.dumps(changelog))
-    # print(patch_resp.json())
+    # create_asset_resp = client.post(
+    #     f"/meta/{dataset}/{version}/assets", json=asset_payload
+    # )
+    # print(json.dumps(create_asset_resp.json(), indent=2))
+    # assert create_asset_resp.json()["status"] == "success"
+    # asset_id = create_asset_resp.json()["data"]["asset_id"]
+
+    # # Verify there are two assets now
+    # get_resp = client.get(f"/meta/{dataset}/{version}/assets")
+    # assert len(get_resp.json()["data"]) == 2
+    #
+    # # Verify the existence of tasks for the new asset
+    # non_default_tasks = client.get(f"/tasks/assets/{asset_id}").json()["data"]
+    # assert len(existing_tasks) == 7
+    # for task in non_default_tasks:
+    #     assert len(task["change_log"]) == 1
+    #     assert task["change_log"][0]["status"] == "pending"
+    #
+    # # Arbitrarily choose a task and add a changelog with status "failed"
+    # sample_task_id = existing_tasks[0]["task_id"]
+    # patch_payload = {
+    #     "change_log": [
+    #         {
+    #             "date_time": "2020-06-25 14:30:00",
+    #             "status": "failed",
+    #             "message": "Womp womp!",
+    #             "detail": "None",
+    #         }
+    #     ]
+    # }
+    # patch_resp = client.patch(f"/tasks/{sample_task_id}", json=patch_payload)
+    # assert patch_resp.json()["status"] == "success"
+    #
+    # # Verify the asset status is now "failed"
+    # get_resp = client.get(f"/meta/{dataset}/{version}/assets/{asset_id}")
+    # assert get_resp.json()["data"]["status"] == "failed"
+    #
+    # # ... but that the version status is still "saved"
+    # get_resp = client.get(f"/meta/{dataset}/{version}")
+    # assert get_resp.json()["data"]["status"] == "saved"
+
```

Comment:
I think, i did not fully implement this. Will take a look and make it a seperate PR.
Also probably best to make this a seperate test